contracts/mocks/MockMarginCalculator.sol

```diff
@@ -14,6 +15,61 @@ contract MockMarginCalculator {
         view
         returns (bool isValid)
     {
+        if (finalVault.shortOtokens.length == 0 || finalVault.shortAmounts.length == 0) return true;
+
+        bool isExcess = getExcessMargin(finalVault, OtokenInterface(finalVault.shortOtokens[0]).collateralAsset());
+
+        return isExcess;
+    }
+
+    function getExcessMargin(MarginAccount.Vault memory _vault, address _denominated)
+        public
+        view
+        returns (bool isExcess)
+    {
+        // ensure the number of collateral, long and short array is valid.
+        _checkIsValidSpread(_vault);
+        // ensure the long asset is valid for the short asset.
+        _checkIsMarginableLong(_vault);
+
         return true;
     }
+
+    function _checkIsValidSpread(MarginAccount.Vault memory _vault) internal pure {
+        require(_vault.shortOtokens.length <= 1, "MarginCalculator: Too many short otokens in the vault.");
+        require(_vault.longOtokens.length <= 1, "MarginCalculator: Too many long otokens in the vault.");
+        require(_vault.collateralAssets.length <= 1, "MarginCalculator: Too many collateral assets in the vault.");
```

Comment:
We should move the checks to controller like you suggested. In the mock contract just use mocked return value without requirement clauses.

Reply:
I'm gonna resolve this given we decide to keep the checks in calculator